actions/run-nextstrain-build-on-aws-batch/action.yaml

```diff
@@ -0,0 +1,68 @@
+name: Run Nextstrain build on AWS Batch
+description: >-
+  This GitHub Actions action is intended to be called by workflows in our other
+  repos when they need to run a nextstrain build on AWS Batch.
+
+inputs:
+  runtime-envvar-names:
+    description: >-
+      A list of environment variable names for environment variables that
+      should be passed to the build. The envvar names should be separated by
+      a space, e.g. "GITHUB_RUN_ID SLACK_TOKEN SLACK_CHANNELS"
+    type: string
+    required: false
+
+  cpus:
+    description: >-
+      Number of CPUs/cores/threads/jobs to utilize at once.
+      Will be passed to the `--cpus` option for `nextstrain build` and the
+      `--cores` option for Snakemake.
+    type: number
+    default: 4
+    required: false
+
+  memory:
+    description: >-
+      Amount of memory to make available to the build. Passed to the
+      `--memory` option for `nextstrain build`.
+    type: string
+    required: false
+
+  build-args:
+    description: >-
+      Additional command-line arguments to pass to `nextstrain build` after
+      the build directory (e.g. to Snakemake).
+    type: string
+    default: ""
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - uses: nextstrain/.github/actions/setup-nextstrain-cli@master
+
+    - name: Creating env.d
+      shell: bash
+      run: |
+        "$GITHUB_ACTION_PATH"/write-envdir env.d \
+          ${{ inputs.runtime-envvar-names }}
```

Comment:
> handling runtime-envvar-names fully as data and never as code

Could you clarify what you mean by this? I'm confused coming back to this now.

Reply:
Use quoting/escaping/etc. to make sure the value of `runtime-envvar-names` is always interpreted as a sequence of string literals instead of potentially as shell metachars, commands, etc.

This sort of quoting isn't enough, since it happens outside the shell:

```suggestion
          "${{ inputs.runtime-envvar-names }}"
```

Instead, passing via the environment is the first step:

```yaml
env:
  varnames: ${{ inputs.runtime-envvar-names }}
```

so that we can do quoting inside the shell:

```suggestion
          "$varnames"
```

which prevents the value being treated as code… but completely breaks the desired use of space-separated values.

To remedy that, we could use the same sort of [YAML-in-YAML workaround](https://github.com/nextstrain/.github/blob/cc6f4385a45bd6ed114ab4840416fd90cc46cd1b/.github/workflows/pathogen-repo-ci.yaml#L47-L64) I have in the pathogen-repo-ci workflow in this repo, e.g. so the input is specified as:

```yaml
- uses: nextstrain/.github/actions/run-nextstrain-build-on-aws-batch@master
  with:
    runtime-envvar-names: |
      - GITHUB_RUN_ID
      - SLACK_TOKEN
      - SLACK_CHANNELS
```

or instead use the simpler word splitting facilities of bash with `read`.

Putting that all together, e.g.:

```yaml
env:
  varnames: ${{ inputs.runtime-envvars-names }}
run: |
  read -a varnames -r -d $'\0' <<<"$varnames"

  "$GITHUB_ACTION_PATH"/write-envdir env.d "${varnames[@]}"
```

Reply:
addressed in #44